### PR TITLE
Add support for brotli decompression of response

### PR DESCRIPTION
**Pull request by coditva**
## PR Checklist:
- [x] I have run `npm test` locally and all tests are passing.
- [x] I have added/updated tests for any new behavior.
       
- [x] If this is a significant change, an issue has already been created where the problem / solution was discussed: https://github.com/postmanlabs/postman-app-support/issues/4091
       

## PR Description

This change adds support for brotli encoding. This can be enabled by setting the options `brotli` to `true`.

### CODE REVIEW

File: lib/brotli.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
+'use strict'
+
+var stream = require('stream')
+var inherit = require('util').inherits
+var brotliDecompressBuffer = require('brotli/decompress')
+var BrotliDecompress
+
+BrotliDecompress = function (options) {
```
Comment (codenirvana) [inline lib/brotli.js]: Use the constructor approach instead.

```js
var Transform = require('stream').Transform

var transformStream = new Transform({
  transform(chunk, encoding, callback) {
    // ...
  }
});
```

File: lib/brotli.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+}
+
+/**
+ * Creates a BrotliDecompress stream
+ *
+ * @param {Object=} options
+ *
+ * @returns {*}
+ */
+module.exports.createBrotliDecompress = function (options) {
```
Comment (codenirvana) [inline lib/brotli.js]:
```suggestion
module.exports.createBrotliDecompress = typeof zlib.createBrotliDecompress === 'function' ?
  zlib.createBrotliDecompress : createBrotliDecompress
  // createBrotliDecompress returns new custom BrotliDecompress instance
```

File: lib/brotli.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
+'use strict'
+
+var Transform = require('stream').Transform
+var brotliDecompressBuffer = require('brotli/decompress')
```
Comment (codenirvana) [inline lib/brotli.js]: Avoid loading this in-memory if not required.

File: lib/brotli.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+var zlib = require('zlib')
+var createBrotliDecompress
+
+createBrotliDecompress = new Transform({
+  transform (chunk, encoding, callback) {
+    try {
+      callback(null, Buffer.from(brotliDecompressBuffer(chunk), encoding))
```
Comment (codenirvana) [inline lib/brotli.js]: `safe-buffer`

File: lib/brotli.js [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+  }
+})
+
+/**
+ * Creates a BrotliDecompress stream
+ * supports faster and native brotli, if available, else fallback to userland module
+ *
+ * @param {Object=} options
+ *
+ * @returns {*}
```
Comment (codenirvana) [inline lib/brotli.js]:
```
* @function
* @param [Object] options Brotli options
* @returns {Object} Returns a new BrotliDecompress object
```

File: request.js [hunk 6]
```diff
@@ -441,8 +442,13 @@ Request.prototype.init = function (options) {
...
 
-  if (self.gzip && !self.hasHeader('accept-encoding')) {
-    self.setHeader('Accept-Encoding', 'gzip, deflate')
+  if (!self.hasHeader('accept-encoding')) {
+    var acceptEncoding = []
+
+    self.gzip && acceptEncoding.push('gzip, deflate')
+    self.brotli && acceptEncoding.push('br')
+
+    acceptEncoding.length && self.setHeader('Accept-Encoding', acceptEncoding.join(', '))
```
Comment (codenirvana) [inline request.js]: Using string concatenation instead of transforming the array.

File: request.js [hunk 7]
```diff
@@ -1226,7 +1232,7 @@ Request.prototype.onRequestResponse = function (response) {
     }
 
     var responseContent
-    if (self.gzip && !noBody(response.statusCode)) {
+    if ((self.gzip || self.brotli) && !noBody(response.statusCode)) {
```
Comment (codenirvana) [inline request.js]: isn't this going to decode `gzip` response even if the `gzip` option is not set?

File: lib/brotli.js (hunk 1 above)
Comment (coditva) [reply lib/brotli.js]: Reverting back because constructor approach doesn't seem to work with browser tests

**Pull request merged**